src/util/HeaderUtil.ts

```diff
@@ -427,6 +436,31 @@ export function parseContentType(contentType: string): { type: string } {
   return { type: contentTypeValue };
 }
 
+/**
+ * Parses the Content-Type header and also parses any parameters in the header.
+ *
+ * @param input - The Content-Type header string.
+ *
+ * @throws {@link BadRequestHttpError}
+ * Thrown on invalid header syntax.
+ *
+ * @returns A {@link ContentType} object containing the value and optional parameters.
+ */
+export function parseContentTypeWithParameters(input: string): ContentType {
```

Comment:
```suggestion
export function parseContentType(input: string): ContentType {
```
I would replace the `parseContentType` function with this so we consistently make use of it everywhere. There are only 2 places where it is used right now anyway, and those should change their behaviour anyway in this PR. This does make this PR a breaking change though which means it should target the next version branch.